src/pkg/scan/postprocessors/report_converters.go

```diff
@@ -0,0 +1,201 @@
+	}
+	rp, err := c.fromSchema(ctx, reportUUID, reportSummary, vulns)
+	if err != nil {
+		return "", err
+	}
+	return rp, nil
+}
+
+func (c *nativeToRelationalSchemaConverter) toSchema(ctx context.Context, reportUUID string, registrationUUID string, digest string, rawReportData string) error {
+
+	var vulnReport vuln.Report
+	err := json.Unmarshal([]byte(rawReportData), &vulnReport)
+	if err != nil {
+		return err
+	}
+	for _, v := range vulnReport.Vulnerabilities {
+		vulnV2 := new(scan.VulnerabilityRecord)
+		vulnV2.CVEID = v.ID
+		vulnV2.Description = v.Description
+		vulnV2.Package = v.Package
+		vulnV2.PackageVersion = v.Version
+		vulnV2.Digest = digest
+		vulnV2.PackageType = "Unknown"
+		vulnV2.Fix = v.FixVersion
+		vulnV2.URL = strings.Join(v.Links, "|")
+		vulnV2.RegistrationUUID = registrationUUID
+		vulnV2.Severity = v.Severity.String()
+		vulnV2.Report = reportUUID
+
+		// process the CVSS scores if the data is available
+		if (vuln.CVSS{} != v.CVSSDetails) {
+			vulnV2.CVE3Score = v.CVSSDetails.ScoreV3
+			vulnV2.CVE2Score = v.CVSSDetails.ScoreV2
+			vulnV2.CVSS3Vector = v.CVSSDetails.VectorV3
+			vulnV2.CVSS2Vector = v.CVSSDetails.VectorV2
+		}
+		if len(v.CWEIds) > 0 {
+			vulnV2.CWEIds = strings.Join(v.CWEIds, ",")
+		}
+
+		// marshall the presented vendor attributes as a json string
+		if len(v.VendorAttributes) > 0 {
+			vendorAttributes, err := json.Marshal(v.VendorAttributes)
+			// set the vendor attributes iff unmarshalling is successful
+			if err == nil {
+				vulnV2.VendorAttributes = string(vendorAttributes)
+			}
+		}
+		_, err = c.dao.InsertForReport(ctx, reportUUID, vulnV2)
+		if err != nil {
+			return err
+		}
+	}
+	log.Infof("Converted %d vulnerability records to the new schema for report ID %s and scanner Id %s", len(vulnReport.Vulnerabilities), reportUUID, registrationUUID)
+	return nil
+}
+
+func (c *nativeToRelationalSchemaConverter) fromSchema(ctx context.Context, reportUUID string, reportSummary string, records []*scan.VulnerabilityRecord) (string, error) {
+	if len(reportSummary) == 0 {
+		return "", nil
+	}
+	vulnerabilityItems := make([]* vuln.VulnerabilityItem,0)
+	for _, record := range records {
+		vi := new(vuln.VulnerabilityItem)
+		vi.ID =  record.CVEID
+		vi.ArtifactDigest = record.Digest
+		vi.CVSSDetails.ScoreV2 = record.CVE2Score
+		vi.CVSSDetails.ScoreV3 = record.CVE3Score
+		vi.CVSSDetails.VectorV2 = record.CVSS2Vector
+		vi.CVSSDetails.VectorV3 = record.CVSS3Vector
+		cweIds := strings.Split(record.CWEIds, ",")
+		for _, cweId := range cweIds {
+			vi.CWEIds = append(vi.CWEIds, cweId)
+		}
+		vi.CWEIds = cweIds
+		vi.Description = record.Description
+		vi.FixVersion = record.Fix
+		vi.Version = record.PackageVersion
+		urls := strings.Split(record.URL, "|")
+		for _, url := range  urls {
+			vi.Links = append(vi.Links, url)
+		}
+		vi.Severity = vuln.ParseSeverityVersion3(record.Severity)
+		vi.Package = record.Package
+		vendorAttributes := make([]vuln.VendorAttribute, 0)
```

Comment:
As we discussed we can use `map[string]interface{}` for vendor attributes rather than a slice of VendorAttribute structs.

Reply:
Done